alpha-build-machinery/make/targets/openshift/crd-schema-gen.mk

```diff
@@ -0,0 +1,25 @@
+CRD_SCHEMA_GEN_VERSION ?=v0.2.1
+CRD_SCHEMA_GEN_APIS ?=./vendor/github.com/openshift/api/operator/v1
+
+# $1 - schema gen version
+# $2 - manifests directory
+# $3 - package APIs
+crd-schema-gen:
+	mkdir -p $(CRD_SCHEMA_GEN_TEMP)/bin
+	git clone -b $(CRD_SCHEMA_GEN_VERSION) --single-branch --depth 1 https://github.com/kubernetes-sigs/controller-tools.git $(CRD_SCHEMA_GEN_TEMP)/gen
+	cd $(CRD_SCHEMA_GEN_TEMP)/gen; GO111MODULE=on go build ./cmd/controller-gen; mv controller-gen ../bin
+	curl -f -L -o $(CRD_SCHEMA_GEN_TEMP)/bin/yq https://github.com/mikefarah/yq/releases/download/2.4.0/yq_$(shell uname -s | tr A-Z a-z)_amd64 && chmod +x $(CRD_SCHEMA_GEN_TEMP)/bin/yq
+.PHONY: crd-schema-gen
+
+update-codegen-crds: CRD_SCHEMA_GEN_TEMP :=$(shell mktemp -d)
+update-codegen-crds: crd-schema-gen
+	$(CRD_SCHEMA_GEN_TEMP)/bin/controller-gen \
+		schemapatch:manifests=./manifests \
+		output:dir=./manifests \
+		paths="$(subst $() $(),;,$(CRD_SCHEMA_GEN_APIS))"
+		shopt -s nullglob; for p in manifests/*.crd.yaml-merge-patch; do $(CRD_SCHEMA_GEN_TEMP)/bin/yq m -i "$${p%%.crd.yaml-merge-patch}.crd.yaml" "$$p"; done
+.PHONY: update-codegen-crds
+
+verify-codegen-crds: update-codegen-crds
+	git diff -q manifests/ || { echo "Changed manifests: "; echo; git diff; false; }
```

Comment:
verify shouldn't require git - generate it into tmp folder and check if it matches what's already generated

Reply:
leads to complexity. Why no git?

Reply:
a) it depends on what you have staged and what not
b) if you add a new file, it may not be tracked
c) git can be in various states

verify shouldn't change the files it is verifying - this overwrites them on verify (not on update) which is certainly not expected

Reply:
controller-gen support to output into another directory. That could be used.